src/pages/GiftDetailPage.jsx

```diff
@@ -1,103 +1,93 @@
 import React from "react";
 import { useParams } from "react-router-dom";
-// import useGift from "../hooks/use-gift";
 import CompareButton from "../components/CompareButton.jsx";
-
+import { Link } from "react-router-dom";
 import { allGifts } from "../data";
 
 function GiftDetailPage() {
-  const { id } = useParams(); // Retrieve the id parameter from the URL
-  const giftData = allGifts.find(g => g.id === parseInt(id));
```

Comment:
Good practice is to use more descriptive variable names rather than single char.
Please consider to change 'g' to something more clear eg: can use 'gift' or 'giftItem' or something else.

Reply:
yep fixed